mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp

```diff
@@ -693,6 +693,8 @@ SPIRVTypeConverter::SPIRVTypeConverter(spirv::TargetEnvAttr targetAttr,
   addConversion([this](IndexType /*indexType*/) { return getIndexType(); });
 
   addConversion([this](IntegerType intType) -> std::optional<Type> {
+    if (!this->options.convertIntAsScalar)
```

Comment:
The use of this hidden option is currently a hack solution, as when we pass in the integer type we fail the linked check, which will then cause us to return a `nullptr`. Proper implementation would presumably ensure that we are passing the compatibility and extension requirements. So hints for how to solve this are needed.

https://github.com/llvm/llvm-project/blame/77c43e14897a404fbf4a132a1a75d49ba2ec08c1/mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp#L232-L235.

Reply:
Can't we resolve this by adding a module with the right target env around the code in test cases? Have you checked where the `const spirv::TargetEnv &targetEnv` parameter in comes from in `convertScalarType`?